src/globus_cli/commands/collection/list.py

```diff
@@ -0,0 +1,115 @@
+import click
+
+from globus_cli.login_manager import LoginManager
+from globus_cli.parsing import command
+from globus_cli.principal_resolver import default_identity_id_resolver
+from globus_cli.services.gcs import get_gcs_client
+from globus_cli.termio import FORMAT_TEXT_TABLE, formatted_print
+
+STANDARD_FIELDS = [
+    ("ID", "id"),
+    ("Display Name", "display_name"),
+    ("Owner", default_identity_id_resolver.field),
+    ("Collection Type", "collection_type"),
+    ("Storage Gateway ID", "storage_gateway_id"),
+]
+
+
+class ChoiceSlugified(click.Choice):
+    """
+    Allow either hyphens or underscores, e.g. both 'mapped-collections' or
+    'mapped_collections'
+    """
+
+    def convert(self, value, param, ctx):
+        if value is None:
+            return None
+        value = value.replace("_", "-")
+        return super().convert(value, param, ctx)
+
+
+def hyphen_to_underscore(ctx, param, value):
+    if not value:
+        return None
+    return [x.replace("-", "_") for x in value]
```

Comment:
Optional refactor: you could fold this into `ChoiceSlugified` and skip using a callback altogether.
Just `super().convert(value, param, ctx).replace("-", "_")` above, and this goes away.

I think I like that better (using a custom param type and no callback vs both), but I don't mind if you prefer to leave it as-is.

Reply:
I kind of suspected it wouldn't be necessary to have both, thanks for the suggestion.

Reply:
FWIW, it also took me a few minutes for me to realize how to do this.

It's actually a case in which the plethora of options `click` gives us is somewhat paralyzing. I was reading `click.ParamType` again to remember all of the things it can -- and can't -- do!